source/core/slang-io.cpp

```diff
@@ -564,9 +564,57 @@ namespace Slang
 
     bool Path::createDirectoryRecursive(const String& path)
     {
-        std::error_code ec;
-        std::filesystem::create_directories(path.getBuffer(), ec);
-        return !ec;
+        String finalPath = Path::simplify(path);
+        if (finalPath.getLength() == 0)
+        {
+            return false;
+        }
+
+        List<String> pathList;
+
+        // Check whether the parent directories exist, and add to the pathList if they are
+        // not, we will create all the directories from back of the list.
+        String parentDir = finalPath;
+        while(1)
```

Comment:
`while(1)` may trigger style warnings in some compilers, in Slang we prefer `for(;;)`.

Reply:
Ah, I see. Will change.
I think I have another place also using that. I will change that as well.